src/annmas/segment/command.py

```diff
@@ -22,22 +37,64 @@ def main(model, threads, output_bam, input_bam):
     logger.info(f"annmas: segment started")
 
     threads = mp.cpu_count() if threads <= 0 or threads > mp.cpu_count() else threads
-    logger.info(f"running with {threads} thread(s)")
+    logger.info(f"Running with {threads} thread(s)")
 
     m = build_default_model()
     if model is not None:
         m.from_yaml(model)
-        logger.info(f"using pretrained annotation model {model}")
+        logger.info(f"Using pretrained annotation model {model}")
     else:
-        logger.info(f"using default annotation model")
+        logger.info(f"Using default annotation model")
+
+    num_reads_annotated = 0
+    num_sections = 0
 
     pysam.set_verbosity(0)  # silence message about the .bai file not being found
-    with pysam.AlignmentFile(input_bam, "rb", check_sq=False, require_index=False) as bf:
-        for read in bf:
-            print(read.query_name)
-            segment_read(read, m)
+    with pysam.AlignmentFile(input_bam, "rb", check_sq=False, require_index=False) as bam_file:
+        with concurrent.futures.ThreadPoolExecutor(max_workers=threads) as executor:
+            future_to_segmented_read = {executor.submit(segment_read, r, m): r for r in bam_file}
+
+            out_bam_header_dict = {'HD': {'VN': '1.0', 'SO': "unknown", 'pb': '>=3.01'}}
+            with pysam.AlignmentFile(output_bam, 'wb', header=out_bam_header_dict) as out_bam_file:
```

Comment:
Shall we just grab the header from the input BAM and supply it here, perhaps with an additional program group?

Reply:
Yeah - had some troubles with the documentation for pysam and headers, but I figured it out.  Update incoming…